src/store.js

```diff
@@ -182,20 +207,23 @@ export function getActiveAndEntryExperimentKeyStates(results, keyStatesLoaded) {
 }
 
 export function setActiveAndEntryKeyStates(version, context, config, configKeyStates, genomeKeyStates) {
-  const loadedKeys = new Set()
-  genomeKeyStates.loaded.forEach(function(key) {
-    loadedKeys.add(key);
-  });
-  configKeyStates.loaded.forEach(function(key) {
-    loadedKeys.add(key);
-  });
-
   const results = evaluatePredicates(version, context, config);
-  configKeyStates.experiments.clear();
 
   results.forEach(function(expResults, eid) {
-    const expKeyStates = new Map();
-    configKeyStates.experiments.set(eid, expKeyStates);
+    const expKeyStates = configKeyStates.experiments.get(eid);
+    if (!expKeyStates) {
+      return;
+    }
+
+    const loadedKeys = new Set();
+    const loadedExpGen = (genomeKeyStates.experiments.get(eid) || new Map()).get('loaded') || new Set();
```

Comment:
Isn't this creating a Map, only to turn around and create a Set? It seems like you should just move this whole block into an if statement?

Reply:
It's creating a map if the experiment map doesnt exist so it can go deeper to see if the loaded set exists. To put this in an if statement feels like overkill as the results are the same.

Reply:
I changed it a little